### DISCUSSION TITLE
Encryption of Data at Rest in Milvus

### DISCUSSION THREAD

**Discussion by fckbo** [milvus-io/milvus#29326] [Q&A and General discussion]
Hello, 
May I ask if someone could confirm is both the underlying etcd and Cloud Object Storage could be encrypted and Milvus still operating properly ? 

Happy to hear if any body has tested this ? of course assuming it would conceptually work.

P.S.
Of course everybody understand that I'm not talking about encryption of data in transit here.

Thx in advance for your answers

**Comment by xiaofan-luan** [comment]
On Zilliz cloud, we enabled object storage encryption and it workes perfectly.

For milvus, data will be in three place, object storage, log storage and local disk(as caching)

For object storage and log storage, you can enable any encrptions because we are retrieving data using kafka S3 API.
For local disk, for now we don't have any data encrptions and that might be a interesting feature to work with

**Reply by rarnold-conga** [reply]
@xiaofan-luan can you clarify if that's still the situation with Milvus today - that the locally cached information on disk is not encrypted? Thanks!

**Reply by xiaofan-luan** [reply]
currently yes, it's still under raw  data format.
if you are using ebs as cache, one of the option is to enable encryption on ebs
